tests/v2_validation/cattlevalidationtest/core/common_fixtures.py

```diff
@@ -2957,7 +2986,8 @@ def add_digital_ocean_hosts(client, count, size="2gb",
     if docker_version == "1.10":
         os_version = "ubuntu-14-04-x64"
     for i in range(0, count):
-        create_args = {"hostname": random_str(),
+        # need extra random names for this one(sorry).
+        create_args = {"hostname": random_str() + str(random_num()),
```

Comment:
I figured out the problem, so basically random library in python uses time in seconds on the running machine to seed the generator, that's why there were failures in parallel jobs, we might put another variable in the random.seed(), in the mean time we added a work around in Jenkins job to sleep random number less < 99 seconds  before starting the job.

**CC** @moelsayed